test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs

```diff
@@ -1796,11 +1802,11 @@ private void MultithreadedHistogramTest<T>(long[] expected, T[] values)
     private class UpdateThreadArguments<T>
         where T : struct, IComparable
     {
-        public ManualResetEvent MreToBlockUpdateThread;
-        public ManualResetEvent MreToEnsureAllThreadsStart;
+        public ManualResetEvent? MreToBlockUpdateThread;
+        public ManualResetEvent? MreToEnsureAllThreadsStart;
         public int ThreadsStartedCount;
-        public Instrument<T> Instrument;
-        public T[] ValuesToRecord;
+        public Instrument<T>? Instrument;
+        public T[]? ValuesToRecord;
     }
```

Comment:
The reason these had an error is because they weren't set by the constructor, so these were all set as nullable to satisfy the compiler.
But this has a consequence of now all the referenced code above needs to perform a null check, which was ignored with the null-forgiving operator (`!`). 

In my opinion, this is messy and changes the intent of the arguments as they weren't originally intended to be null.

Alternatively, we could change the `private class UpdateThreadArguments<T>` to a `struct` which satisfies the original error and doesn't introduce consequences. I may be overthinking this because this is just a test class, but I think that would be a cleaner change. Eager to get a second opinion here.

Reply:
I have made some direct changes. Please let me know if you are fine with this.

Reply:
LGTM. Thanks